Default the plotting unit to None

xiki-tempula opened the pull request:
Fix #247
Default the unit in `plot_dF_state` , `plot_convergence` and `plot_ti_dhdl` to None.
Remove the array input in `plot_convergence`.

I have fixed the issue here @DrDomenicoMarson

orbeckst commented on src/alchemlyb/visualisation/convergence.py:
```diff
@@ -5,37 +5,35 @@
-    `forward`: A list of free energy estimate from the first X% of data,
-    where `forward_error` is the corresponding error.
+    :func:`~alchemlyb.convergence.forward_backward_convergence` or
+    :func:`~alchemlyb.convergence.fwdrev_cumavg_Rc`. The input should be a
```
(from other PR on Rc but that's fine)

orbeckst reviewed (changes requested):
Looking good but please add entry in CHANGES under Changes

xiki-tempula commented:
@orbeckst Thanks for the review.
I have added the change log

> Is there a test that shows that (1) None works and that (2) changing units works?

I have added this as well.

orbeckst commented:
thanks!
